rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -289,10 +289,10 @@ public synchronized int joinLRA(StringBuilder recoveryUrl, URL lra, long timeLim
                                     String compensatorData) {
         if (lra ==  null) {
             lraTrace(null, "Error missing LRA header in join request");
+        } else {
+            lraTrace(lra, "join LRA");
```

Comment:
hi @mmusgrov, if you don't mind I have one more doubt on here on the `LRAService`. From the check it seems the `lra` variable could be `null` but then the lra is used in way it should not be null (see https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java#L309). I refer to the call `participant = transaction.enlistParticipant(lra, ...)` where `enlistParticipant` provides the `lra` to the next method which calls `coordinatorUrl.toExternalForm()` and could end with NPE possibly.

Reply:
I added the trace method during the early prototype and can now be safely deleted

Reply:
@mmusgrov ok, I did it so. Thank you for the explanation.

Reply:
@ochaloup I meant that the lra_trace method can now be removed from the source file. Altrnatively just add the null check at line 340:

> if (lraId != null && lras.containsKey(lraId)) {

Reply:
@mmusgrov I see, thanks for the hint. I think having the trace log helps during issue investigation. I think your alternative approach is better.

On top of this my worry was about `lra` being null at call `enlistParticipant` (https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java#L309). If `lra` is null it leaks to the method and will `NPE` will be thrown somewhere during execution. What to throw NPE directly here? https://github.com/jbosstm/narayana/blob/master/rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java#L291